java/arcs/android/devtools/DevWebSocketFactory.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.devtools
+
+import arcs.core.util.TaggedLog
+import fi.iki.elonen.NanoHTTPD
+import fi.iki.elonen.NanoWSD
+import fi.iki.elonen.NanoWSD.WebSocketFrame.CloseCode
+import java.io.IOException
+
+/**
+ * An extension of [NanoWSD] for devtools to connect Arcs to a remote device for debugging.
+ */
+class DevWebSocketFactory {
+
+    fun getDevWebSocket() = devWebSocket
+
+    companion object devWebSocket : NanoWSD("localhost", 33317) {
+        private var wsdSocket: WsdSocket? = null
+        private var started = false
+        private val log = TaggedLog { "DevWebSocket" }
+
+        /**
+         * Send a string to the client.
+         */
+        fun send(msg: String) {
+            if (wsdSocket?.open ?: false) {
+                wsdSocket?.send(msg)
+            } else {
```

Comment:
Pretty sure AndroidStudio will warn/lint/otherwise-complain about this. I suggest resolving any of those issues in AndroidStudio for this PR..